src/cloudflare/internal/images-api.ts

```diff
@@ -303,6 +327,165 @@ function chainStreams<T>(streams: ReadableStream<T>[]): ReadableStream<T> {
   return outputStream;
 }
 
+function concatUint8Arrays(a: Uint8Array, b: Uint8Array): Uint8Array {
+  const result = new Uint8Array(a.length + b.length);
+  result.set(a, 0);
+  result.set(b, a.length);
+  return result;
+}
+
+class Base64Error extends Error {
+  public constructor(cause: unknown) {
+    if (cause instanceof Error) {
+      super(`base64 error: ${cause.message}`, { cause });
+    } else {
+      super('unknown base64 error');
+    }
+  }
+}
+
+function createBase64EncoderTransformStream(
+  maxEncodeChunkSize: number = 32 * 1024 + 1
+): TransformStream<Uint8Array, Uint8Array> {
+  if (maxEncodeChunkSize % 3 != 0) {
+    // Ensures that chunks don't require padding
+    throw new Error('maxChunkSize must be a multiple of 3');
+  }
+
+  let buffer: Uint8Array | null = null;
+  const asciiEncoder = new TextEncoder();
+
+  const toBase64: (buf: Uint8Array) => Uint8Array = (buf) => {
+    const binaryString = String.fromCharCode.apply(null, Array.from(buf));
+
+    const base64String = btoa(binaryString);
+    const base64Bytes = asciiEncoder.encode(base64String);
+
+    return base64Bytes;
+  };
+
+  return new TransformStream<Uint8Array, Uint8Array>({
+    transform(chunk, controller): void {
+      const currentData = buffer ? concatUint8Arrays(buffer, chunk) : chunk;
+      buffer = null;
+
+      let offset = 0;
+
+      while (currentData.length - offset >= maxEncodeChunkSize) {
+        const sliceToEnd = offset + maxEncodeChunkSize;
+        const processChunk = currentData.slice(offset, sliceToEnd);
+        offset = sliceToEnd;
+
+        try {
+          controller.enqueue(toBase64(processChunk));
+        } catch (error) {
+          controller.error(new Base64Error(error));
+          buffer = null;
+          return;
+        }
+      }
+
+      buffer = offset < currentData.length ? currentData.slice(offset) : null;
+    },
+
+    flush(controller): void {
+      if (buffer && buffer.length > 0) {
+        try {
+          controller.enqueue(toBase64(buffer));
+        } catch (error) {
+          const errMsg = error instanceof Error ? error.message : String(error);
+          controller.error(new Error(`base64 encoding error: ${errMsg}`));
+        }
+      }
+      buffer = null;
+    },
+  });
+}
+
+function createBase64DecoderTransformStream(
+  maxChunkSize: number = 32 * 1024
+): TransformStream<Uint8Array, Uint8Array> {
+  if (maxChunkSize % 4 !== 0 || maxChunkSize <= 0) {
+    throw new Error('maxChunkSize must be a positive multiple of 4.');
+  }
+
+  let base64Buffer: Uint8Array | null = null;
+  const asciiDecoder = new TextDecoder('ascii');
```

Comment:
You can create this in the global scope. No need to create this everytime this method is executed.

Reply:
While true, it's generally discouraged to create shared `TextDecoder` instances because they can contain state depending on what encoding is used and how they are utilized. Keeping this here makes sense I think.